src/main/java/chalmers/pimp/controller/components/ImageChooser.java

```diff
@@ -43,8 +48,34 @@ public PixelData openDialog(Window window) throws IOException {
       System.out.println("No image file was selected!");
       return null;
     }
+    setResentFileName(file);
 
     Image image = ImageImportService.importImage(file);
     return FXToPixelDataService.createPixelDataCopy(image);
   }
+
+  /**
+   * Sets the resentFileName for this ImageChooser, based on the supplied {@code File}. Removes the
+   * extension.
+   *
+   * @param resentFile the {@code File} from which the name comes from.
+   */
+  private void setResentFileName(File resentFile) {
+    List<String> temp = new ArrayList<>(Arrays.asList(resentFile.getName().split("\\.")));
+    temp.remove((temp.size() - 1));
+    StringBuilder returnString = new StringBuilder();
+    for (String s : temp) {
+      returnString.append(s);
+    }
+    resentFileName = returnString.toString();
```

Comment:
This means "image.png" => "image" which is correct but it also means "image.firstimage.1.png" => "imagefirstimage1" which is false.

Reply:
I recommend to insert this code from line 65
```
return resentfile.getName().replace("." + temp.getLast(), ""));
```